### PR TITLE
python link verification

### PR DESCRIPTION
**Pull request by KarishmaGhiya**

### CODE REVIEW

File: sdk/core/azure/CHANGELOG.md [hunk 1]
```diff
@@ -507,7 +505,7 @@ version before installing the latest release.
 * Azure Storage has moved.
     https://github.com/Azure/azure-storage-python
 
-    http://azure-storage.readthedocs.org/en/latest/
+    https://azure.github.io/azure-sdk-for-python/ref/Storage.html
```
Comment (lmazuel) [inline sdk/core/azure/CHANGELOG.md:508]: I don't see the point of changing a link from a Changelog written in 2015

File: sdk/customproviders/azure-mgmt-customproviders/README.md [hunk 2]
```diff
@@ -2,14 +2,7 @@
...
 This package has been tested with Python 2.7, 3.5, 3.6, 3.7 and 3.8.
-For a more complete view of Azure libraries, see the [Github repo](https://github.com/Azure/azure-sdk-for-python/sdk)
-
-
-# Usage
-
-For code examples, see [CustomProviders Management](https://docs.microsoft.com/python/api/overview/azure/customproviders)
-on docs.microsoft.com.
-
+For a more complete view of Azure libraries, see the [Github repo](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk)
```
Comment (lmazuel) [inline sdk/customproviders/azure-mgmt-customproviders/README.md:5]: https://aka.ms/azsdk/releases/all

Comment (lmazuel) [reply sdk/customproviders/azure-mgmt-customproviders/README.md:5]: Or just point to the root like most of the pacakges

File: tools/azure-sdk-tools/packaging_tools/templates/README.md [hunk 3]
```diff
@@ -7,8 +7,7 @@ For a more complete view of Azure libraries, see the [Github repo](https://githu
 
 # Usage
 
-For code examples, see [{{package_pprint_name}}](https://docs.microsoft.com/python/api/overview/azure/{{package_doc_id}})
-on docs.microsoft.com.
+This is a template project purely for engineering use and doesn't have code examples on docs.ms.com.
```
Comment (lmazuel) [inline tools/azure-sdk-tools/packaging_tools/templates/README.md:10]: This is incorrect, this is a template that is used automatically to generate code. Please revert this change.

File: sdk/core/azure/CHANGELOG.md (hunk 1 above)
Comment (rakshith91) [inline sdk/core/azure/CHANGELOG.md:508]: is this necessary?

File: sdk/hybridkubernetes/azure-mgmt-hybridkubernetes/README.md [hunk 4]
```diff
@@ -4,13 +4,6 @@ This is the Microsoft Azure HybridKubernetes Management Client Library.
 This package has been tested with Python 2.7, 3.5, 3.6, 3.7 and 3.8.
 For a more complete view of Azure libraries, see the [Github repo](https://github.com/Azure/azure-sdk-for-python/)
```
Comment (rakshith91) [inline sdk/hybridkubernetes/azure-mgmt-hybridkubernetes/README.md:5]: question: some of these point to https://github.com/Azure/azure-sdk-for-python/tree/master/sdk
and some to the root https://github.com/Azure/azure-sdk-for-python - is there any specific pattern? - don't see much value in changing these for consistency, so don't bother changing them all

File: sdk/machinelearning/azure-mgmt-machinelearningcompute/README.md [hunk 5]
```diff
@@ -15,12 +15,6 @@ library.
 For a more complete set of Azure libraries, see the
 [azure](https://pypi.python.org/pypi/azure) bundle package.
```
Comment (rakshith91) [inline sdk/machinelearning/azure-mgmt-machinelearningcompute/README.md:16]: https://pypi.python.org/pypi/azure package is deprecated - this link should be replaced with https://aka.ms/azsdk/python/all

File: sdk/servicebus/azure-servicebus/CHANGELOG.md [hunk 6]
```diff
@@ -141,7 +141,7 @@ Version 7.0.0b1 is a preview of our efforts to create a client library that is u
 
 * Introduces new AMQP-based API.
 * Original HTTP-based API still available under new namespace: azure.servicebus.control_client
-* For full API changes, please see updated [reference documentation](https://docs.microsoft.com/python/api/overview/azure/servicebus/client?view=azure-python).
+* For full API changes, please see updated [reference documentation](https://docs.microsoft.com/en-us/python/api/azure-servicebus/azure.servicebus?view=azure-python).
```
Comment (rakshith91) [inline sdk/servicebus/azure-servicebus/CHANGELOG.md:144]: I think it's better to remove en-us/ from the link

File: sdk/timeseriesinsights/azure-mgmt-timeseriesinsights/README.md [hunk 7]
```diff
@@ -15,12 +15,6 @@ library.
 For a more complete set of Azure libraries, see the
 [azure](https://pypi.python.org/pypi/azure) bundle package.
```
Comment (rakshith91) [inline sdk/timeseriesinsights/azure-mgmt-timeseriesinsights/README.md:16]: same as above - you might want to use replace all :)

File: tools/azure-devtools/doc/scenario_base_tests.md [hunk 8]
```diff
@@ -84,12 +84,12 @@ its `__init__` method (probably using `super`):
 * `recording_processors` and `replay_processors`:
   Lists of `RecordingProcessor` instances for making changes to requests and responses
   during test recording and test playback, respectively.
-  See [recording_processors.py](src/azure_devtools/scenario_tests/recording_processors.py)
+  See [recording_processors.py](https://github.com/Azure/azure-sdk-for-python/blob/e6d329933b00855bb5873fcfa7abe97a0f03c44a/tools/azure-devtools/src/azure_devtools/scenario_tests/recording_processors.py)
```
Comment (rakshith91) [inline tools/azure-devtools/doc/scenario_base_tests.md:87]: is this correct? if so, why this specific commit in the link?

File: sdk/hybridkubernetes/azure-mgmt-hybridkubernetes/README.md (hunk 4 above)
Comment (KarishmaGhiya) [reply sdk/hybridkubernetes/azure-mgmt-hybridkubernetes/README.md:5]: I am not sure, these were already there so i updated only the ones which are broken. i am not sure what the standard should be

File: sdk/timeseriesinsights/azure-mgmt-timeseriesinsights/README.md (hunk 7 above)
Comment (KarishmaGhiya) [reply sdk/timeseriesinsights/azure-mgmt-timeseriesinsights/README.md:16]: i think all the packages have this reference - should i update in all 73 packages?

File: tools/azure-sdk-tools/packaging_tools/templates/README.md (hunk 3 above)
Comment (KarishmaGhiya) [reply tools/azure-sdk-tools/packaging_tools/templates/README.md:10]: @scbedd  should we update this to -  https://docs.microsoft.com/python/api/azure-{{package_doc_id}}  
The base link is incorrect - https://docs.microsoft.com/python/api/overview/azure/

File: sdk/timeseriesinsights/azure-mgmt-timeseriesinsights/README.md (hunk 7 above)
Comment (rakshith91) [reply sdk/timeseriesinsights/azure-mgmt-timeseriesinsights/README.md:16]: yes

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 9]
```diff
@@ -275,9 +275,9 @@ revamped configuration API. Static `create_config` methods have been renamed
 
 ### New features:
 - Added credentials for authenticating users:
-[`DeviceCodeCredential`](https://azure.github.io/azure-sdk-for-python/ref/azure.identity.html#azure.identity.DeviceCodeCredential),
-[`InteractiveBrowserCredential`](https://azure.github.io/azure-sdk-for-python/ref/azure.identity.html#azure.identity.InteractiveBrowserCredential),
-[`UsernamePasswordCredential`](https://azure.github.io/azure-sdk-for-python/ref/azure.identity.html#azure.identity.UsernamePasswordCredential)
+[`DeviceCodeCredential`](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-identity/1.2.0/azure.identity.html#azure.identity.DeviceCodeCredential),
```
Comment (chlowell) [inline sdk/identity/azure-identity/CHANGELOG.md:278]: This is the changelog for the pre-GA beta 2. We seem not to have docs for it anymore, which I think is okay because it's a year old and unsupported. So although the current links 404, is linking to documentation of a later, significantly different release really an improvement?

Comment (KarishmaGhiya) [reply sdk/identity/azure-identity/CHANGELOG.md:278]: Is it ok if i remove these links entirely?

File: sdk/machinelearning/azure-mgmt-machinelearningcompute/README.md (hunk 5 above)
Comment (kristapratico) [reply sdk/machinelearning/azure-mgmt-machinelearningcompute/README.md]: isn't this README generated, though? should this be an update in autorest?

Comment (rakshith91) [reply sdk/machinelearning/azure-mgmt-machinelearningcompute/README.md]: you are right - we need to update in autorest too - but replacing few links now is less costlier than regenrating all the packages - that said, we need to fix in autorest

File: sdk/identity/azure-identity/CHANGELOG.md (hunk 9 above)
Comment (KarishmaGhiya) [reply sdk/identity/azure-identity/CHANGELOG.md]: I have removed the links from these

File: sdk/machinelearning/azure-mgmt-machinelearningcompute/README.md (hunk 5 above)
Comment (KarishmaGhiya) [reply sdk/machinelearning/azure-mgmt-machinelearningcompute/README.md]: can you please let me know what file in autorest needs to be updated?

Comment (KarishmaGhiya) [reply sdk/machinelearning/azure-mgmt-machinelearningcompute/README.md]: Is it this - https://github.com/Azure/azure-sdk-for-python/blob/8447aadb71cdaad69524785efd0dff18960d3076/tools/azure-sdk-tools/packaging_tools/templates/README.md#L3

File: sdk/core/azure/CHANGELOG.md (hunk 1 above)
Comment (KarishmaGhiya) [reply sdk/core/azure/CHANGELOG.md]: removed the link

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 10]
```diff
@@ -209,7 +209,7 @@ the Azure CLI's client ID will be used.
   `client_id`
   - transport configuration is now done through keyword arguments as
   described in
-  [`azure-core` documentation](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/docs/configuration.md)
+  [`azure-core` documentation](https://github.com/Azure/azure-sdk-for-python/blob/azure-identity_1.0.0b4/sdk/core/azure-core/docs/configuration.md)
```
Comment (chlowell) [inline sdk/identity/azure-identity/CHANGELOG.md:212]:
```suggestion
  [`azure-core` documentation](https://github.com/Azure/azure-sdk-for-python/blob/azure-identity_1.0.0/sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md#transport)
```

File: sdk/compute/azure-mgmt-imagebuilder/azure/mgmt/imagebuilder/models/_models.py [hunk 11]
```diff
@@ -539,23 +539,23 @@ def __init__(self, **kwargs):
 
 class ImageTemplatePlatformImageSource(ImageTemplateSource):
     """Describes an image source from [Azure Gallery
-    Images](https://docs.microsoft.com/en-us/rest/api/compute/virtualmachineimages).
+    Images](https://docs.microsoft.com/rest/api/compute/virtualmachineimages).
```
Comment (lmazuel) [inline sdk/compute/azure-mgmt-imagebuilder/azure/mgmt/imagebuilder/models/_models.py:542]: Changing generated code has no value, since the next generation will put it back again.
I would suggest to make a list of mgmt packages that have those kind of issues, and open Swagger bugs instead.

For instance, the root cause of this one is here:
https://github.com/Azure/azure-rest-api-specs/blob/900ef530bda427a6ba9f61ac6f64952ea7650a37/specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2020-02-14/imagebuilder.json#L579

File: sdk/core/azure-servicemanagement-legacy/tests/recordings/test_legacy_mgmt_misc.test_add_data_disk_from_disk_name.yaml [hunk 12]
```diff
@@ -223,7 +223,7 @@ interactions:
         \ Foundation Server 2013 Update 4 on Windows Server 2012 R2</Label><Location>East\
         \ Asia;Southeast Asia;Australia East;Australia Southeast;Brazil South;North\
         \ Europe;West Europe;Japan East;Japan West;Central US;East US;East US 2;North\
-        \ Central US;South Central US;West US</Location><LogicalSizeInGB>128</LogicalSizeInGB><Name>03f55de797f546a1b29d1b8d66be687a__Team-Foundation-Server-2013-Update4-WS2012R2</Name><OS>Windows</OS><Eula>http://www.microsoft.com/en-us/download/details.aspx?id=13350</Eula><Description>Microsoft\
+        \ Central US;South Central US;West US</Location><LogicalSizeInGB>128</LogicalSizeInGB><Name>03f55de797f546a1b29d1b8d66be687a__Team-Foundation-Server-2013-Update4-WS2012R2</Name><OS>Windows</OS><Eula>http://www.microsoft.com/download/details.aspx?id=13350</Eula><Description>Microsoft\
```
Comment (lmazuel) [inline sdk/core/azure-servicemanagement-legacy/tests/recordings/test_legacy_mgmt_misc.test_add_data_disk_from_disk_name.yaml:226]: Don't update YAML files, those are recordings of actual call for the service

File: sdk/compute/azure-mgmt-imagebuilder/azure/mgmt/imagebuilder/models/_models.py (hunk 11 above)
Comment (KarishmaGhiya) [reply sdk/compute/azure-mgmt-imagebuilder/azure/mgmt/imagebuilder/models/_models.py:542]: Thanks i'll make these changes in this PR - https://github.com/Azure/azure-sdk-for-python/pull/12697

File: sdk/core/azure-servicemanagement-legacy/tests/recordings/test_legacy_mgmt_misc.test_add_data_disk_from_disk_name.yaml (hunk 12 above)
Comment (KarishmaGhiya) [reply sdk/core/azure-servicemanagement-legacy/tests/recordings/test_legacy_mgmt_misc.test_add_data_disk_from_disk_name.yaml:226]: Thanks i'll make these changes in this PR - #12697

File: tools/azure-sdk-tools/packaging_tools/templates/README.md (hunk 3 above)
Comment (KarishmaGhiya) [reply tools/azure-sdk-tools/packaging_tools/templates/README.md]: reverted this

File: sdk/datafactory/azure-mgmt-datafactory/azure/mgmt/datafactory/models/_models_py3.py [hunk 13]
```diff
@@ -15409,7 +15409,11 @@ class HDInsightOnDemandLinkedService(LinkedService):
     :type zookeeper_node_size: object
     :param script_actions: Custom script actions to run on HDI ondemand
      cluster once it's up. Please refer to
+<<<<<<< HEAD
```
Comment (lmazuel) [inline sdk/datafactory/azure-mgmt-datafactory/azure/mgmt/datafactory/models/_models_py3.py:15412]: That's not OK :(

Comment (KarishmaGhiya) [reply sdk/datafactory/azure-mgmt-datafactory/azure/mgmt/datafactory/models/_models_py3.py:15412]: :(

File: sdk/core/azure-servicemanagement-legacy/tests/recordings/test_legacy_mgmt_misc.test_add_data_disk_from_disk_name.yaml [hunk 14]
```diff
@@ -1748,28 +1748,28 @@ interactions:
         \ West;Central US;East US;East US 2;North Central US;South Central US;West\
         \ US</Location><LogicalSizeInGB>32</LogicalSizeInGB><Name>9b7cfe55da0349d5a8316a8cc4741f87__StorSimple-VirtualAppliance-1100-20140710</Name><OS>Windows</OS><Description>Microsoft\
         \ StorSimple Virtual Appliance 1100</Description><ImageFamily>StorSimple Virtual\
-        \ Appliance 1100</ImageFamily><ShowInGui>false</ShowInGui><PublishedDate>2014-07-09T00:41:59Z</PublishedDate><IsPremium>false</IsPremium><PrivacyUri>http://azure.microsoft.com/en-us/support/legal/privacy-statement/</PrivacyUri><PublisherName>Microsoft\
+        \ Appliance 1100</ImageFamily><ShowInGui>false</ShowInGui><PublishedDate>2014-07-09T00:41:59Z</PublishedDate><IsPremium>false</IsPremium><PrivacyUri>http://azure.microsoft.com/support/legal/privacy-statement/</PrivacyUri><PublisherName>Microsoft\
```
Comment (lmazuel) [inline sdk/core/azure-servicemanagement-legacy/tests/recordings/test_legacy_mgmt_misc.test_add_data_disk_from_disk_name.yaml:1751]: There should be no changes in any YAML file at all in this PR

File: sdk/servicebus/azure-servicebus/CHANGELOG.md (hunk 6 above)
Comment (KarishmaGhiya) [reply sdk/servicebus/azure-servicebus/CHANGELOG.md:167]: I am doing that in a separate PR

Comment (KarishmaGhiya) [reply sdk/servicebus/azure-servicebus/CHANGELOG.md:167]:
```suggestion
* For full API changes, please see updated [reference documentation](https://docs.microsoft.com/python/api/azure-servicebus/azure.servicebus?view=azure-python).
```

File: tools/azure-devtools/doc/scenario_base_tests.md (hunk 8 above)
Comment (KarishmaGhiya) [inline tools/azure-devtools/doc/scenario_base_tests.md:87]:
```suggestion
  See [recording_processors.py](https://github.com/Azure/azure-sdk-for-python/blob/master/tools/azure-devtools/src/azure_devtools/scenario_tests/recording_processors.py)
```

File: tools/azure-devtools/doc/scenario_base_tests.md [hunk 15]
```diff
@@ -84,12 +84,12 @@ its `__init__` method (probably using `super`):
...
   Lists of `RecordingProcessor` instances for making changes to requests and responses
   during test recording and test playback, respectively.
-  See [recording_processors.py](src/azure_devtools/scenario_tests/recording_processors.py)
+  See [recording_processors.py](https://github.com/Azure/azure-sdk-for-python/blob/master/tools/azure-devtools/src/azure_devtools/scenario_tests/recording_processors.py)
   for some examples and how to implement them.
 * `recording_patches` and `replay_patches`:
   Lists of patches to apply to functions, methods, etc.
   during test recording and playback, respectively.
-  See [patches.py](src/azure_devtools/scenario_tests/patches.py)
+  See [patches.py](https://github.com/Azure/azure-sdk-for-python/blob/39f2757d504babfa49a70ced87a3150a57b0f462/tools/azure-devtools/src/azure_devtools/scenario_tests/patches.py)
```
Comment (KarishmaGhiya) [inline tools/azure-devtools/doc/scenario_base_tests.md:92]:
```suggestion
  See [patches.py](https://github.com/Azure/azure-sdk-for-python/blob/master/tools/azure-devtools/src/azure_devtools/scenario_tests/patches.py)
```

File: sdk/identity/azure-identity/CHANGELOG.md [hunk 16]
```diff
@@ -284,9 +284,9 @@ revamped configuration API. Static `create_config` methods have been renamed
 
 ### New features:
 - Added credentials for authenticating users:
-[`DeviceCodeCredential`](https://azure.github.io/azure-sdk-for-python/ref/azure.identity.html#azure.identity.DeviceCodeCredential),
-[`InteractiveBrowserCredential`](https://azure.github.io/azure-sdk-for-python/ref/azure.identity.html#azure.identity.InteractiveBrowserCredential),
-[`UsernamePasswordCredential`](https://azure.github.io/azure-sdk-for-python/ref/azure.identity.html#azure.identity.UsernamePasswordCredential)
+ - `DeviceCodeCredential`
```
Comment (lmazuel) [inline sdk/identity/azure-identity/CHANGELOG.md:287]: @chlowell can you sign of of that?

File: sdk/identity/azure-identity/CHANGELOG.md (hunk 10 above)
Comment (chlowell) [reply sdk/identity/azure-identity/CHANGELOG.md:212]: This new link doesn't cover transport customization. The one I suggested does, and I believe it's correct for azure-identity 1.0.0b4.

File: sdk/identity/azure-identity/CHANGELOG.md (hunk 16 above)
Comment (chlowell) [reply sdk/identity/azure-identity/CHANGELOG.md:287]: Works for me, we don't have docs for this version (1.0.0b2) anymore and I prefer removing these to adding links to later documentation.

File: sdk/core/azure/CHANGELOG.md (hunk 1 above)
Comment (KarishmaGhiya) [reply sdk/core/azure/CHANGELOG.md]: I removed these links entirely

File: sdk/keyvault/azure-keyvault-secrets/CHANGELOG.md [hunk 17]
```diff
@@ -52,7 +52,7 @@ detail the new keyword arguments):
 - `Secret` now has attribute `properties`, which holds certain properties of the
 secret, such as `version`. This changes the shape of the returned `Secret` type,
 as certain properties of `Secret` (such as `version`) have to be accessed
-through the `properties` property. See the updated [docs](https://azure.github.io/azure-sdk-for-python/ref/azure.keyvault.secrets.html)
+through the `properties` property. See the updated [docs](https://azure.github.io/azure-sdk-for-python/ref/Keyvault.html#azure-keyvault-secrets)
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-secrets/CHANGELOG.md:55]: Whoops, I missed this one:
```suggestion
through the `properties` property.
```

File: sdk/keyvault/azure-keyvault-keys/CHANGELOG.md [hunk 18]
```diff
@@ -70,7 +70,7 @@ been renamed to `KeyCurveName`, `KeyOperation`, and `KeyType`, respectively.
 - `Key` now has attribute `properties`, which holds certain properties of the
 key, such as `version`. This changes the shape of the returned `Key` type,
 as certain properties of `Key` (such as `version`) have to be accessed
-through the `properties` property. See the updated [docs](https://azure.github.io/azure-sdk-for-python/ref/azure.keyvault.keys.html)
+through the `properties` property.
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-keys/CHANGELOG.md:73]: Sorry to keep picking on these Key Vault files but  now we have an orphaned "for details." line after each of these because PR suggestions can't include unchanged lines 😞

Comment (KarishmaGhiya) [reply sdk/keyvault/azure-keyvault-keys/CHANGELOG.md:73]: removed them

**Comment by weshaggard** [comment]
Some pipelines were hitting the github rate limiting issue and aren't real failures, so overriding.
